monai/transforms/utility/dictionary.py

```diff
@@ -1320,18 +1321,28 @@ def __init__(
         image_key: str | None = None,
         image_threshold: float = 0.0,
         output_shape: Sequence[int] | None = None,
+        max_indices_per_class: int | None = None,
         allow_missing_keys: bool = False,
     ) -> None:
         super().__init__(keys, allow_missing_keys)
         self.indices_postfix = indices_postfix
         self.image_key = image_key
+        self.max_indices_per_class = max_indices_per_class
         self.converter = ClassesToIndices(num_classes, image_threshold, output_shape)
 
     def __call__(self, data: Mapping[Hashable, Any]):
         d = dict(data)
         image = d[self.image_key] if self.image_key else None
         for key in self.key_iterator(d):
-            d[str(key) + self.indices_postfix] = self.converter(d[key], image)
+            out = self.converter(d[key], image)
```

Comment:
if it's feasible in general, perhaps we consider subsampling when generating the indices?

https://github.com/Project-MONAI/MONAI/blob/9e33ff25f34ebd11e026c49634743d9389564765/monai/transforms/utils.py#L332-L341

Reply:
I'm open to any option, I just need to limit the amount of cache here.  feel free to edit, or how would I like me to proceed? thanks

Reply:
sure I'll try to update `map_classes_to_indices` for the same purpose.

Reply:
thank you!